### PR TITLE
Feature/tp 103025

### PR DESCRIPTION
**Pull request by IvarKoreman-Ritense**
#103025 - [BE] Endpoint to get tasks containing custom column values

### CODE REVIEW

File: process-document/src/main/kotlin/com/ritense/processdocument/domain/CaseTask.kt
Comment (Klaas-Ritense) [inline process-document/src/main/kotlin/com/ritense/processdocument/domain/CaseTask.kt:1]: Missing license header on this file and some others

File: process-document/src/test/java/com/ritense/processdocument/web/TaskListResourceTest.kt [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            .andExpect(MockMvcResultMatchers.jsonPath("$.content").isArray)
+            .andExpect(MockMvcResultMatchers.jsonPath("$.content[*].id", hasItems(*taskList.map { it.id }.toTypedArray())))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.content[0]", hasKey("name")))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.content[0]", hasKey("assignee")))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.content[0]", hasKey("created")))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.content[0]", hasKey("due")))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.content[0]", hasKey("owner")))
+    }
+
+    private fun startNewProcessedWithTestData(): List<Map<String, String>> {
```
Comment (Klaas-Ritense) [inline process-document/src/test/java/com/ritense/processdocument/web/TaskListResourceTest.kt]:
```suggestion
    private fun startNewProcessesWithTestData(): List<Map<String, String>> {
```

File: process-document/src/test/java/com/ritense/processdocument/web/TaskListResourceIntTest.kt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+    @Autowired
+    lateinit var taskService: TaskService
+
+    @BeforeEach
+    fun setUp() {
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build()
+
+        val allProcessIds = runtimeService.createProcessInstanceQuery().list().map { it.id }
+        runtimeService.deleteProcessInstances(allProcessIds, "Cleaning for tests", true, false)
```
Comment (Klaas-Ritense) [inline process-document/src/test/java/com/ritense/processdocument/web/TaskListResourceIntTest.kt:76]: Minor: We usually do this by adding `@Transactional` above the class. Spring will then make sure all database changes are rolled back after the test

File: process-document/src/main/kotlin/com/ritense/processdocument/service/CaseTaskListSearchService.kt [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    private val DOC_PREFIX = "doc:"
+    private val CASE_PREFIX = "case:"
+    private val DOCUMENT_FIELD_MAP: Map<String, String> = mapOf(
+        "definitionId.name" to "documentDefinitionId.name",
+        "definitionId.version" to "documentDefinitionId.key",
+        INTERNAL_STATUS to INTERNAL_STATUS_ORDER
+    )
+
+    fun getTasksByCaseDefinition(caseDefinitionName: String, assignmentFilter: CamundaTaskService.TaskFilter, pageable: Pageable): Page<TaskListRowDto> {
+        // No authorization on this level, as we have to fully rely on the documentSearchService for filtering results
```
Comment (Klaas-Ritense) [inline process-document/src/main/kotlin/com/ritense/processdocument/service/CaseTaskListSearchService.kt]: documentSearchService?

Im missing logic to filter the task list. Users should only see tasks that they have permission to. This is also done for the existing task list. see `CamundaTaskService.findTasksFiltered(..)`

File: process-document/src/test/java/com/ritense/processdocument/web/TaskListResourceIntTest.kt (hunk 2 above)
Comment (IvarKoreman-Ritense) [reply process-document/src/test/java/com/ritense/processdocument/web/TaskListResourceIntTest.kt:76]: I've just tried this, but unfortunately my test break at the `assertThat(taskList).hasSize(testUserData.size)` when multiple tests are ran.

File: process-document/src/main/kotlin/com/ritense/processdocument/service/CaseTaskListSearchService.kt (hunk 3 above)
Comment (IvarKoreman-Ritense) [reply process-document/src/main/kotlin/com/ritense/processdocument/service/CaseTaskListSearchService.kt]: Added the PBAC filtering.

**Review by Klaas-Ritense — APPROVED** [review]

**Review by marijnritense — APPROVED** [review]

**Pull request merged**